src/tools/miri/src/concurrency/stack.rs

```diff
@@ -0,0 +1,47 @@
+use crate::*;
+
+/// An encapsulated call stack, so encapsulated to enable efficient caching of metadata about the
+/// contained `Frame`.
```

Comment:
This seems to give out mutable access to basically the entire thing; in which sense is this 'encapsulated'?